ipaserver/install/dnskeysyncinstance.py

```diff
@@ -164,6 +165,87 @@ def __setup_dnssec_containers(self):
 
         self._ldap_mod("dnssec.ldif", {'SUFFIX': self.suffix, })
 
+    def __are_named_options_configured(self, options):
+        """
+        Check whether sysconfig of named is patched or not.
+        Additional command line options for named are passed
+        via OPTIONS env variable. Since custom options can be
+        supplied by a vendor, at least, the base parsing of such
+        is required.
+        Current named command line options:
+        NS_MAIN_ARGS "46A:c:C:d:D:E:fFgi:lL:M:m:n:N:p:P:sS:t:T:U:u:vVx:X:"
+        If there are several same options the last passed wins.
+        """
+        if options:
+            pattern = r"[ ]*-[a-zA-Z46]*E[ ]*(.*?)(?: |$)"
+            p = re.compile(pattern)
+            engines = re.findall(p, options)
+            if engines and engines[-1] == constants.NAMED_OPENSSL_ENGINE:
+                return True
+
+        return False
+
+    def setup_named_openssl_conf(self):
+        if constants.NAMED_OPENSSL_ENGINE is not None:
+            logger.debug("Setup OpenSSL config for BIND")
+            # setup OpenSSL config for BIND,
+            # this one is needed because FreeIPA installation
+            # disables p11-kit-proxy PKCS11 module
+            conf_file_dict = {
+                'OPENSSL_ENGINE': constants.NAMED_OPENSSL_ENGINE,
+                'SOFTHSM_MODULE': paths.LIBSOFTHSM2_SO,
+            }
+            named_openssl_txt = ipautil.template_file(
+                os.path.join(paths.USR_SHARE_IPA_DIR,
+                             "bind.openssl.cnf.template"),
+                conf_file_dict)
+            with open(paths.DNSSEC_OPENSSL_CONF, 'w') as f:
+                os.fchmod(f.fileno(), 0o644)
+                f.write(named_openssl_txt)
```

Comment:
Which user and group needs read access to the file?

Reply:
current ownership and permissions:
```
# ls -la /etc/ipa/dnssec/
total 28
drwxr-xr-x 1 root root 4096 Aug 10 09:33 .
drwxr-xr-x 1 root root 4096 Aug 10 09:34 ..
-r--r----- 1 root ods   524 Aug 10 09:33 ipa-dnskeysyncd.keytab
-rw-r--r-- 1 root root  267 Aug 10 09:33 openssl.cnf
-rw-r--r-- 1 root root  145 Aug 10 09:33 softhsm2.conf
-r-------- 1 root root   30 Aug 10 09:33 softhsm_pin_so
```

Consumers of this file are:
- ipa-dnskeysyncd
- named/named-pkcs11

```
systemctl cat ipa-dnskeysyncd
# /usr/lib/systemd/system/ipa-dnskeysyncd.service
[Unit]
Description=IPA key daemon

[Service]
EnvironmentFile=/etc/sysconfig/ipa-dnskeysyncd
ExecStart=/usr/libexec/ipa/ipa-dnskeysyncd
User=ods
Group=named
SupplementaryGroups=ods
PrivateTmp=yes
...
```

```
# systemctl cat named
# /usr/lib/systemd/system/named.service
[Unit]
Description=Berkeley Internet Name Domain (DNS)
Wants=nss-lookup.target
Wants=named-setup-rndc.service
Before=nss-lookup.target
After=named-setup-rndc.service
After=network.target

[Service]
Type=forking
Environment=NAMEDCONF=/etc/named.conf
EnvironmentFile=-/etc/sysconfig/named
Environment=KRB5_KTNAME=/etc/named.keytab
PIDFile=/run/named/named.pid

ExecStartPre=/bin/bash -c 'if [ ! "$DISABLE_ZONE_CHECKING" == "yes" ]; then /usr/sbin/named-che
ckconf -z "$NAMEDCONF"; else echo "Checking of zone files is disabled"; fi'
ExecStart=/usr/sbin/named -u named -c ${NAMEDCONF} $OPTIONS
ExecReload=/bin/sh -c 'if /usr/sbin/rndc null > /dev/null 2>&1; then /usr/sbin/rndc reload; els
e /bin/kill -HUP $MAINPID; fi'

ExecStop=/bin/sh -c '/usr/sbin/rndc stop > /dev/null 2>&1 || /bin/kill -TERM $MAINPID'

PrivateTmp=true
...
```

It seems that below would enough:
```
# ls -la /etc/ipa/dnssec/openssl.cnf 
-rw-r----- 1 root named 267 Aug 10 09:33 /etc/ipa/dnssec/openssl.cnf
```